geode-core/src/test/java/org/apache/geode/management/internal/beans/stats/GCStatsMonitorTest.java

```diff
@@ -33,16 +45,94 @@
 
   @Before
   public void setUp() {
-    gcStatsMonitor = new GCStatsMonitor(testName.getMethodName());
+    ValueMonitor valueMonitor = mock(ValueMonitor.class);
+    gcStatsMonitor = new GCStatsMonitor(testName.getMethodName(), valueMonitor);
+
     assertThat(gcStatsMonitor).isNotNull();
-    assertThat(gcStatsMonitor.getStatistic("collections")).isEqualTo(0L);
-    assertThat(gcStatsMonitor.getStatistic("collectionTime")).isEqualTo(0L);
+    assertThat(gcStatsMonitor.getCollections()).isEqualTo(0L);
+    assertThat(gcStatsMonitor.getCollectionTime()).isEqualTo(0L);
   }
 
   @Test
   public void getStatisticShouldReturnZeroForUnknownStatistics() {
     assertThat(gcStatsMonitor.getStatistic("unknownStatistic")).isEqualTo(0);
   }
 
+  @Test
+  public void addStatsToMonitor() throws Exception {
+    Statistics stats = mock(Statistics.class);
+    when(stats.getUniqueId()).thenReturn(11L);
+    StatisticDescriptor d1 = mock(StatisticDescriptor.class);
+    when(d1.getName()).thenReturn(StatsKey.VM_GC_STATS_COLLECTIONS);
+    StatisticDescriptor d2 = mock(StatisticDescriptor.class);
+    when(d2.getName()).thenReturn(StatsKey.VM_GC_STATS_COLLECTION_TIME);
+    StatisticDescriptor[] descriptors = {d1, d2};
+    StatisticsType type = mock(StatisticsType.class);
+    when(stats.getType()).thenReturn(type);
+    when(type.getStatistics()).thenReturn(descriptors);
+
+    when(stats.get(any(StatisticDescriptor.class))).thenReturn(8L, 300L);
```

Comment:
This way of specifying return values unnecessarily couples the test to the current implementation—it insists that the implementation must call `stats.get(sd)` this many times, and in this specific order.

On this line and the two later ones, it would be better to associate each value with the appropriate statistic descriptor, the way the other test does.